githubmarkdownui/ci.py

```diff
@@ -34,13 +42,20 @@ class CIJob:
     executed within a build pipeline."""
     metadata: CIJobMetadata
     tasks: List[CITask]
+    child_jobs: Optional[List[CIJob]] = None
 
+    def ci_task_list(tasks: Optional[List[CITask]] = None) -> str:
```

Comment:
I think it would be useful to have the option to filter out which tasks you want to display, so someone could make a `CIJob` object with all the available tasks for a job, but if they only wanted to display a subset of them they could pass them in here instead of being forced to make a new `CIJob` object with only the relevant tasks

Reply:
That feels a little complicated to use and to implement. For example, how would we validate tasks passed in all belong to the `CIJob` object.

But I see your point. An alternative way to support similar requirement is - we can pass in some flag, like:status. When given, we should show tasks with that status

Reply:
I agree, it would be best to avoid the validation. I've added an enum `CIStatus` which has either `SUCCEEDED` or `FAILED` options, and the `ci_task_list` function takes in an optional parameter for status now. It also makes sense to use `CIStatus` in the classes instead of emoji to signal success or failure so I've changed that as well.